### ISSUE TITLE
Enabling manual remote administration of Vagrant boxes sans default creds

### ISSUE THREAD

**Issue by syreal17** [picoCTF/picoCTF#283]
There's multiple issues with the usage of default Vagrant creds (u:vagrant/p:vagrant) as they are the main way in which the Vagrant web box gets problem metadata from the Vagrant shell box. Issue #38 addresses that security problem. This issue (issue #283) aims to establish separate SSH credentials for developers/deployers so that both boxes can be remotely administered directly after "vagrant up" with a clear password or key that is not predictable to outsiders. This thought process was inspired by issue #147.

With local access to the virtual machines (VM's), "vagrant ssh shell" is probably preferred (which I believe uses a vagrant-generated key pair and not password), but local access should perhaps not be the only method of access, esp. using infrastructure other than our own.

See https://www.vagrantup.com/docs/boxes/base.html#default-user-settings for more information about Vagrant default config.

Since addressing this issue is part of preparing for issue #38, I will opt to create a key pair for a new user "admin" that will be present on the Vagrant boxes, so that developers/deployers can maintain remote access to the picoCTF platform without using default vagrant credentials.

**Comment by carlislemc** [comment]
Keep in mind the admin web interface has these creds to ssh to the shell to enable problems

**Comment by syreal17** [comment]
Indeed, I'm creating remote admin creds other than vagrant/vagrant, switching over web server shell access to those, then disabling the vagrant/vagrant login.

**Comment by syreal17** [comment]
I've successfully used Ansible to procedurally generate a password for an "admin" user that it creates on the Vagrant shell box. The plaintext of the password is stored in a file on the shell box. Tomorrow, I will start to investigate how to "exfiltrate" that file from the VM to the host which seems to be an altogether better place for that file.

I've seen a straight forward way in Ansible to generate a private key for a system user, but still working on implementing that for our platform.
